runtime/runtime/src/config.rs

```diff
@@ -356,6 +342,40 @@ pub fn total_prepaid_gas(actions: &[Action]) -> Result<Gas, IntegerOverflowError
     Ok(total_gas)
 }
 
+/// Calculates a maximum expected gas price increase during the execution of the transaction.
+///
+/// Note: this is no longer used with ProtocolFeature::ReducedGasRefunds
+fn pessimistic_gas_price(
+    gas_price: u128,
+    sender_is_receiver: bool,
+    fees: &std::sync::Arc<near_parameters::RuntimeFeesConfig>,
+    prepaid_gas: u64,
+) -> Result<u128, IntegerOverflowError> {
+    // If signer is equals to receiver the receipt will be processed at the same block as this
+    // transaction. Otherwise it will processed in the next block and the gas might be inflated.
+    let initial_receipt_hop = if sender_is_receiver { 0 } else { 1 };
+    // The pessimistic gas pricing is a best-effort limit which can be breached in case of
+    // congestion when receipts are delayed before they execute. Hence there is not much
+    // value to tie this limit to the function call base cost. Making it constant limits
+    // overcharging to 6x, which was the value before the cost increase.
+    let minimum_new_receipt_gas = 4_855_842_000_000; // 4.855TGas.
+    // In case the config is free, we don't care about the maximum depth.
+    let receipt_gas_price = if gas_price == 0 {
+        0
+    } else {
+        let maximum_depth =
+            if minimum_new_receipt_gas > 0 { prepaid_gas / minimum_new_receipt_gas } else { 0 };
+        let inflation_exponent = u8::try_from(initial_receipt_hop + maximum_depth)
+            .map_err(|_| IntegerOverflowError {})?;
+        safe_gas_price_inflated(
+            gas_price,
+            fees.pessimistic_gas_price_inflation_ratio,
```

Comment:
Q: this PR modifies both the inflation rate to be 100% (i.e. no inflation) and special cases this code to not be called at all if the protocol feature is enabled. Are both of these changes necessary at the same time? Maybe it'd be possible to get rid of the parameter changes altogether?

Reply:
Yes, the parameter changes could be removed. But then the RPC when queried for the config will report 1.03 as an inflation ratio, without any hints that this is actually depreciated. Like this, it will report a value for new protocol configs that are "correct". (I would deprecate or even remove the parameter, if I knew of a good way of doing it. I can't remember if we ever did that.)

On the other hand, I initially though I would only do the parameter change, without the code changes. But then, I saw you had great progress with limited replayability! :D That's basically why I special case this code to not be called at all. Makes it easy to remove dead code when we no longer versions with pessimistic gas pricing. Otherwise, it's likely to stay in the code base forever.